addons/point_of_sale/static/src/app/screens/partner_list/partner_editor/partner_editor.xml

```diff
@@ -51,18 +51,18 @@
                 <div class="partner-detail col">
                     <label class="form-label label" for="language">Language</label>
                     <select class="detail form-select" t-att="{'disabled': isFieldCommercialAndPartnerIsChild('lang')}" id="language" name="lang" t-model="changes.lang" t-att-class="{'border-danger': props.missingFields.includes('lang')}">
-                        <t t-foreach="pos.langs" t-as="lang" t-key="lang.id">
+                        <t t-foreach="pos.models['res.lang'].getAll()" t-as="lang" t-key="lang.id">
                             <option t-att-value="lang.code" t-att-selected="changes.lang ? ((lang.code === changes.lang) ? true : undefined) : lang.code === pos.user.lang? true : undefined">
                                 <t t-esc="lang.name" />
                             </option>
                         </t>
                     </select>
                 </div>
-                <div t-if="pos.pricelists.length gt 1" class="partner-detail col">
+                <div t-if="pos.models['product.pricelist'].length gt 1" class="partner-detail col">
                     <label class="form-label label" for="pricelist">Pricelist</label>
                     <select class="detail form-select" t-att="{'disabled': isFieldCommercialAndPartnerIsChild('property_product_pricelist')}" id="pricelist" name="property_product_pricelist" t-model="changes.property_product_pricelist" t-att-class="{'border-danger': props.missingFields.includes('property_product_pricelist')}">
-                        <t t-foreach="pos.pricelists" t-as="pricelist" t-key="pricelist.id">
-                            <option t-att-value="pricelist.id" t-att-selected="props.partner.property_product_pricelist ? (pricelist.id === props.partner.property_product_pricelist[0] ? true : undefined) : pricelist.id === pos.default_pricelist.id ? true : undefined">
+                        <t t-foreach="pos.models['product.pricelist'].getAll()" t-as="pricelist" t-key="pricelist.id">
+                            <option t-att-value="pricelist.id" t-att-selected="props.partner.property_product_pricelist ? (pricelist.id === props.partner.property_product_pricelist[0] ? true : undefined) : pricelist.id === pos.config.pricelist_id?.id ? true : undefined">
```

Comment:
```suggestion
                            <option t-att-value="pricelist.id" t-att-selected="props.partner.property_product_pricelist ? (pricelist.id === props.partner.property_product_pricelist?.id ? true : undefined) : pricelist.id === pos.config.pricelist_id?.id ? true : undefined">
```

`props.partner` is already a related record, right? So the pricelist should also be already a record. Double check.

Reply:
Yes nice !